src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import React from 'react'
+import { STATIC_EXECUTION_CONTEXT } from '../constants'
+import _GlobalStyle from '../models/GlobalStyle'
+import StyleSheet from '../models/StyleSheet'
+import { StyleSheetConsumer } from '../models/StyleSheetManager'
+import determineTheme from '../utils/determineTheme'
+import { ThemeConsumer, type Theme } from '../models/ThemeProvider'
+import type { CSSConstructor, Interpolation, Stringifier } from '../types'
+import hashStr from '../vendor/glamor/hash'
+
+export default (stringifyRules: Stringifier, css: CSSConstructor) => {
+  const GlobalStyle = _GlobalStyle(stringifyRules)
+
+  const createGlobalStyle = (
+    strings: Array<string>,
+    ...interpolations: Array<Interpolation>
+  ) => {
+    const rules = css(strings, ...interpolations)
+    const id = `sc-global-${hashStr(JSON.stringify(rules))}`
+    const style = new GlobalStyle(rules, id)
+
+    class GlobalStyleComponent extends React.Component<*, *> {
+      componentWillMount() {
```

Comment:
This can probably become `constructor`

Reply:
Ok. I managed to move the `insert` and `update` of the styling into the render method. So we are only left with `componentWillUnmount` which, I think, is safe to use. Right?